src/ui/submission.rs

```diff
@@ -111,6 +112,21 @@ fn print_final_result(ui: &mut Ui<impl RP>, submission_info: &SubmissionInfo) ->
     Ok(())
 }
 
+pub fn spinner(ui: &mut Ui<impl RP>, c: char) -> Result<char> {
```

Comment:
This likely shouldn't be public. Maybe take `&mut char` instead of returning it. Overall I don't know if relying on the character to track state is the best design, but it's fine as long as a spinner isn't needed elsewhere.
```suggestion
fn spinner(ui: &mut Ui<impl RP>, c: &mut char) -> Result<()> {
```